addon/route-handlers/shorthands/base.js

```diff
@@ -50,48 +53,30 @@ export default class BaseShorthandRouteHandler {
         id = parseInt(request.params.id, 10);
       }
     }
-
     return id;
   }
 
-  _getUrlForRequest(request) {
-    return request && request.url;
-  }
-
-  _getModelNameFromUrl(url, hasId) {
-    let [ urlSplit ] = url.split('?');
-    let path = urlSplit.split('/');
-    path = path[path.length - 1] === '' ? path.slice(0, path.length - 1) : path; // when trailing slash
-    let typePath = hasId ? path[path.length - 2] : path[path.length - 1];
-    let modelName = dasherize(camelize(singularize(typePath)));
-
-    return modelName;
-  }
-
   _getJsonBodyForRequest(request) {
```

Comment:
If we're moving the call to `normalize` here this method name isn't really accurate anymore, can we change it to `getJsonApiDocForRequest`?